compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -1191,11 +1191,14 @@ impl<'a, 'tcx> Visitor<'tcx> for FindInferSourceVisitor<'a, 'tcx> {
                 have_turbofish,
             } = args;
             let generics = tcx.generics_of(generics_def_id);
-            if let Some(argument_index) = generics
+            if let Some(mut argument_index) = generics
                 .own_substs(substs)
                 .iter()
                 .position(|&arg| self.generic_arg_contains_target(arg))
             {
+                if generics.parent.is_none() && generics.has_self {
```

Comment:
Is the situation `generics.parent.is_some() && generics.has_self` possible?
What happens in that case?

Reply:
It shouldn't ever happen. I'd be fine with adding an assertion here, or just remove the check for `generics.parent.is_none()` altogether.

Reply:
I couldn't conjure up such a case either. `own_substs` also checks for `parent` and so do a lot of other places that query `has_self`. I just followed suit.

I can remove the check.